kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManagerImpl.java

```diff
@@ -128,20 +150,35 @@ private RegisteredSchema getSchemaFromSchemaVersion(
     String actualSubject =
         subject.orElse(getSchemaSubjectUnsafe(topicName, isKey, subjectNameStrategy));
 
-    Schema schema =
-        schemaRegistryClient.getByVersion(
-            actualSubject, schemaVersion, /* lookupDeletedSchema= */ false);
-
-    ParsedSchema parsedSchema =
-        EmbeddedFormat.forSchemaType(schema.getSchemaType())
-            .getSchemaProvider()
-            .parseSchema(schema.getSchema(), schema.getReferences(), /* isNew= */ false)
-            .orElseThrow(
-                () ->
-                    Errors.invalidSchemaException(
-                        String.format(
-                            "Error when fetching schema by version. subject = %s, version = %d",
-                            actualSubject, schemaVersion)));
+    Schema schema;
+    try {
+      schema =
+          schemaRegistryClient
+              .get()
+              .getByVersion(actualSubject, schemaVersion, /* lookupDeletedSchema= */ false);
+    } catch (RuntimeException restClientException) {
+      throw new BadRequestException(
+          String.format(
+              "Schema does not exist for subject: %s, version: %s", actualSubject, schemaVersion),
+          restClientException);
+    }
+
+    ParsedSchema parsedSchema;
+    try {
+      parsedSchema =
+          EmbeddedFormat.forSchemaType(schema.getSchemaType())
+              .getSchemaProvider() // throws unsupportedOperationException for bad config
```

Comment:
We should minimize the context of the try-block. Something like:

```
SchemaProvider schemaProvider;
try {
  schemaProvider = ...;
} catch (...) { ... }

ParsedSchema parsedSchema = schemaProvider.parseSchema(...)...
```

Reply:
OK, but then we need two try blocks instead - both getSchemaProvider and parseSchema throw exceptions we want to catch - the orElseThrow just throws if the value provided to it is null if I understand the javadoc correctly.

It seems tidier to have the catch together.  

No strong feelings though, let me know if you want me to split it out to be what you have above, but with a try catch around the parseSchema line too.